src/systemctl/systemctl.c

```diff
@@ -5668,6 +5713,12 @@ static int enable_unit(int argc, char *argv[], void *userdata) {
 
                 polkit_agent_open_if_enabled();
 
+                if (STR_IN_SET(verb, "mask", "unmask")) {
+                        r = unit_exists(*names, verb);
+                        if (r != 0)
+                                return r;
```

Comment:
yeah this could happen . But If a unit A is dependent on B  that is some service requires by A is provided by B. So installing A also B get installed. But A won't work as expected since B is masked already.